webanno-ui-annotation/src/main/java/de/tudarmstadt/ukp/clarin/webanno/ui/annotation/detail/LayerSelector.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2015
+ * Ubiquitous Knowledge Processing (UKP) Lab and FG Language Technology
+ * Technische Universität Darmstadt
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package de.tudarmstadt.ukp.clarin.webanno.ui.annotation.detail;
+
+import de.tudarmstadt.ukp.clarin.webanno.api.annotation.model.AnnotatorState;
+import de.tudarmstadt.ukp.clarin.webanno.model.AnnotationLayer;
+import de.tudarmstadt.ukp.clarin.webanno.ui.annotation.component
+    .DeleteOrReplaceAnnotationModalPanel;
+import org.apache.wicket.ajax.AjaxRequestTarget;
+import org.apache.wicket.ajax.form.AjaxFormComponentUpdatingBehavior;
+import org.apache.wicket.extensions.ajax.markup.html.modal.ModalWindow;
+import org.apache.wicket.markup.html.basic.Label;
+import org.apache.wicket.markup.html.form.ChoiceRenderer;
+import org.apache.wicket.markup.html.form.DropDownChoice;
+
+import java.util.List;
+
+import static de.tudarmstadt.ukp.clarin.webanno.ui.annotation.detail.AnnotationDetailEditorPanel
+    .handleException;
+
+class LayerSelector
+    extends DropDownChoice<AnnotationLayer>
+{
+    private static final long serialVersionUID = 2233133653137312264L;
+
+    LayerSelector(AnnotationFeatureForm featureForm, String aId, List<? extends AnnotationLayer>
+        aChoices)
+    {
+        super(aId, aChoices);
+        setOutputMarkupId(true);
+        setChoiceRenderer(new ChoiceRenderer<>("uiName"));
+        add(new AjaxFormComponentUpdatingBehavior("change")
+        {
+            private static final long serialVersionUID = 5179816588460867471L;
+
+            @Override
+            protected void onUpdate(AjaxRequestTarget aTarget)
+            {
+                AnnotationDetailEditorPanel editorPanel = featureForm.getEditorPanel();
```

Comment:
It seems like the ADEP is needed in this method, because an ADEP specific method is called at line 103:
```
editorPanel.clearFeatureEditorModels(aTarget);
```
 Also, the ADEP is passed as an argument at line 80 to the DeleteOrReplaceAnnotationModalPanel instance what has to be also of type ADEP because the DeleteOrReplaceAnnotationModalPanel instance calls ADEP specific methods.